overlord/registrystate/registrymgr.go

```diff
@@ -52,18 +70,95 @@ func Manager(st *state.State, hookMgr *hookstate.HookManager, _ *state.TaskRunne
 
 func (m *RegistryManager) Ensure() error { return nil }
 
-func setupRegistryHook(st *state.State, snapName, hookName string, ignoreError bool) *state.Task {
-	hookSup := &hookstate.HookSetup{
-		Snap:        snapName,
-		Hook:        hookName,
-		Optional:    true,
-		IgnoreError: ignoreError,
+func (m *RegistryManager) doCommitTransaction(t *state.Task, _ *tomb.Tomb) (err error) {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	tx, _, err := GetStoredTransaction(t)
+	if err != nil {
+		return err
 	}
-	summary := fmt.Sprintf(i18n.G("Run hook %s of snap %q"), hookName, snapName)
-	task := hookstate.HookTask(st, summary, hookSup, nil)
-	return task
+
+	registryAssert, err := assertstateRegistry(st, tx.RegistryAccount, tx.RegistryName)
+	if err != nil {
+		return err
+	}
+	schema := registryAssert.Registry().Schema
+
```

Comment:
we probably want to set t.SetStatus(DoneStatus), to avoid re-running this

Reply:
I've merged this so I can open a PR that depends on this but I'll open a follow-up depending on what we need to do about this